src/util/safelywritablefile.cpp

```diff
@@ -263,6 +243,75 @@ bool SafelyWritableFile::commit() {
     return true;
 }
 
+bool SafelyWritableFile::useTempFileWithPrefix() {
+    if (!QFileInfo(m_origFileName).isWritable()) {
+        kLogger.warning()
+                << "Failed to prepare file for writing:"
+                << m_origFileName
+                << "is not writable.";
+        return false;
+    }
+
+    QString origFilePath = QFileInfo(m_origFileName).canonicalPath();
+    QString origFileCompleteBasename = QFileInfo(m_origFileName).completeBaseName();
+    QString origFileSuffix = QFileInfo(m_origFileName).suffix();
+    QString tempFileCompleteBasename = kSafelyWritableTempFilePrefix +
+            origFileCompleteBasename + '.' + origFileSuffix;
+    QString tempFileName = origFilePath + '/' + tempFileCompleteBasename;
+    m_tempFileName = tempFileName;
+
+    QFile origFile(m_origFileName);
+    QFile tempFile(m_tempFileName);
+
+    if (tempFile.exists() && !tempFile.remove()) {
```

Comment:
I am afraid this deletes any file that has by luck already the temp name. this must not happen.

Reply:
I have deleted it.